packages/react-native/React/Fabric/Mounting/ComponentViews/View/RCTViewComponentView.mm

```diff
@@ -775,6 +783,58 @@ - (void)invalidateLayer
     _filterLayer.zPosition = CGFLOAT_MAX;
     [self.layer addSublayer:_filterLayer];
   }
+  
+  [self clearExistingGradientLayers];
+  auto background = _props->background;
+  if (background.empty()) {
+    return;
+  }
+  for (const auto &gradientPrimitive : background) {
+    CAGradientLayer *gradientLayer = [CAGradientLayer layer];
+    NSMutableArray *colors = [NSMutableArray array];
+    NSMutableArray *locations = [NSMutableArray array];
+    for (const auto& colorStop : gradientPrimitive.colorStops) {
+      if (colorStop.position.has_value()) {
+        auto location = @(colorStop.position.value());
+        UIColor* color = RCTUIColorFromSharedColor(colorStop.color);
+        [colors addObject:(id) color.CGColor];
+        [locations addObject:location];
+      }
+    }
```

Comment:
will this just not add some colors if they do not have a position? in what cases would that happen? i.e. what cases do we have a color with an empty position

Reply:
> what cases do we have a color with an empty position

That should not happen. In view config processors we make sure that position is always set. We set [default position ](https://github.com/facebook/react-native/blob/491c3e9d76a1bc08c1709222cf03259c33128459/packages/react-native/Libraries/StyleSheet/__tests__/processBackgroundImage-test.js#L194) if user doesn't pass a position.